core/trino-main/src/main/java/io/trino/sql/planner/plan/LimitNode.java

```diff
@@ -97,6 +101,17 @@ public boolean isPartial()
         return partial;
     }
 
+    public boolean isOrderSensitive()
+    {
+        return inputOrdering.isPresent();
+    }
+
+    @JsonProperty
+    public Optional<OrderingScheme> getInputOrdering()
```

Comment:
It seems unnecessary to track the input ordering. If the node is order-sensitive, it's sensitive to whatever order the source produces. Tracking the input ordering would only make sense if we're trying to describe a high-level property that the input must satisfy so that when the node is lowered to a more physical form, an appropriate operation (e.g, a Sort) is introduced if the input is not already sorted.

Reply:
We require the input ordering for validation checker ValidateLimitWithPresortedInput.
After applying review comments, I'm also using it in optimizer rules.
E.g.: Allow pushdown through window if the ordering scheme is compatible with the ordering scheme of the window function, pushdown through mark distinct if ordering scheme does not reference the marker column etc.
I'm not sure what's the alternative mechanism to implement above things without tracking input ordering.